Source/bmalloc/libpas/src/libpas/bmalloc_heap_inlines.h

```diff
@@ -90,18 +97,22 @@ static PAS_ALWAYS_INLINE void* bmalloc_try_allocate_auxiliary_zeroed_inline(
     size_t size,
     pas_allocation_mode allocation_mode)
 {
-    return (void*)pas_allocation_result_zero(
-        bmalloc_try_allocate_auxiliary_impl(heap_ref, size, 1, allocation_mode),
-        size).begin;
+    pas_allocation_result result;
+    result = bmalloc_try_allocate_auxiliary_impl(heap_ref, size, 1, allocation_mode);
+    if (allocation_mode == pas_non_compact_allocation_mode && MAR_isAddressInQualifyingPage((void*) result.begin))
```

Comment:
PGM is enabled only when it is  activated. But this profiling is always enabled. I think it is too much. We should do it only when it is activated probabilistically.